test/test_interpreter.cpp

```diff
@@ -17,12 +17,52 @@
 #include "../src/xsystem.hpp"
 #include "../src/xmagics/os.hpp"
 
+#include <iostream>
+
 #include <fstream>
 #if defined(__GNUC__) && !defined(XEUS_CPP_EMSCRIPTEN_WASM_BUILD)
     #include <sys/wait.h>
     #include <unistd.h>
 #endif
 
+/**
+ * @class StreamRedirectRAII
+ * @brief A RAII class to redirect a stream to a stringstream.
+ *
+ * This class redirects the output of a given std::ostream to a std::stringstream.
+ * The original stream is restored when the object is destroyed.
+ */
+class StreamRedirectRAII {
+public:
+    /**
+     * @brief Constructor that starts redirecting the given stream.
+     * @param stream The stream to redirect.
+     */
+    StreamRedirectRAII(std::ostream& stream) : old_stream_buff(stream.rdbuf()), stream_to_redirect(stream) {
+        stream_to_redirect.rdbuf(ss.rdbuf());
+    }
+
+    /**
+     * @brief Destructor that restores the original stream.
+     */
+    ~StreamRedirectRAII() {
+        stream_to_redirect.rdbuf(old_stream_buff);
+    }
+
+    /**
+     * @brief Get the output that was written to the stream.
+     * @return A string containing the output that was written to the stream.
+     */
+    std::string getCaptured() {
+        return ss.str();
+    }
+
+private:
+    std::streambuf* old_stream_buff; ///< The original buffer of the stream.
```

Comment:
```suggestion
    /// The original buffer of the stream.
    std::streambuf* old_stream_buff; 
```

Documenting the public api is generally enough as the expectation is that the private api will vary over time more and will more likely cause the documentation to go out of sync.